### DISCUSSION TITLE
gatsby build - window is not defined

### DISCUSSION THREAD

**Discussion by asztandar** [gatsbyjs/gatsby#36986] [Help]
Hello,
I have a problem with gatsby build - window is not defined.
![image](https://user-images.githubusercontent.com/10097355/200578969-544787ae-84d3-425f-b086-b39b4a2587b2.png)

I don't use "window" anywhere in my code, so it's probably used in some dependency.
Tried using gatsby-node.js:
`exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
    if (stage === "build-html" || stage === "develop-html") {
      actions.setWebpackConfig({
        module: {
          rules: [
            {
              test: /react-openlayers/,
              use: loaders.null(),
            },
          ],
        },
      })
    }
  }`
  
  But I am getting another error:
  
![image](https://user-images.githubusercontent.com/10097355/200580193-eac17eff-9902-462c-98fe-27c02d988bd7.png)

I also tried:
`exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
    if (stage === "build-html" || stage === "develop-html") {
      actions.setWebpackConfig({
        module: {
          rules: [
            {
              test: /react-openlayers|ol|@mui|react-icons|@gatsbyjs|common-tags/,
              use: loaders.null(),
            },
          ],
        },
      })
    }
  }`
  
  but it doesn't work.
  
  Anyone have an idea how to fix the bug and do the build?

**Comment by violy** [comment]
It's seem a common — and famous — mistake, [have you read this documentation section _« How to check if `window` is defined »_](https://www.gatsbyjs.com/docs/debugging-html-builds/#how-to-check-if-window-is-defined) ?
